modules/fedimint-ln-client/src/lib.rs

```diff
@@ -1888,94 +1893,105 @@ impl LightningClientModule {
         }
     }
 
-    pub async fn wait_for_ln_payment(
+    /// Subscribes to either a internal or external lightning payment and
+    /// returns `LightningPaymentOutcome` that indicates if the payment was
+    /// successful or not.
+    pub async fn await_outgoing_payment(
```

Comment:
I think this is a useful function to have, but to give users the most timely feedback subscribing to either of the two `subscribe_*_pay` functions is still necessary. What do you think about unifying these instead and then building the simpler await function on top that only returns the final outcome?

Why I think the subscribe functions are still necessary: e.g. when refunding we know the payment failed 1 consensus round before we got our money back, so if we wait for the entire operation to finish before giving the user feedback we are increasing latency.